core/diirt/diirt-plugins/org.csstudio.diirt.util.core.preferences/src/org/csstudio/diirt/util/core/preferences/DIIRTPreferences.java

```diff
@@ -909,6 +970,110 @@ private static void forceDelete ( final File file ) throws IOException {
+
+        if ( StringUtils.isBlank(s) ) {
+            return s;
+        }
+
+        StringBuilder sb = new StringBuilder(s.length() + 16);
+
+        for ( int i = 0; i < s.length(); i++ ) {
+
+            char c = s.charAt(i);
+
+            switch ( c ) {
+                case ' ':
+                    sb.append("%20");
+                    break;
+                case '!':
+                    sb.append("%21");
+                    break;
+                case '#':
+                    sb.append("%23");
+                    break;
+                case '$':
+                    sb.append("%24");
+                    break;
+                case '%':
+                    sb.append("%25");
+                    break;
+                case '&':
+                    sb.append("%26");
+                    break;
+                case '\'':
+                    sb.append("%27");
+                    break;
+                case '(':
+                    sb.append("%28");
+                    break;
+                case ')':
+                    sb.append("%29");
+                    break;
+                case '*':
+                    sb.append("%2A");
+                    break;
+                case '+':
+                    sb.append("%2B");
+                    break;
+                case ',':
+                    sb.append("%2C");
+                    break;
+                case '<':
+                    sb.append("%3C");
+                    break;
+                case '=':
+                    sb.append("%3D");
+                    break;
+                case '>':
+                    sb.append("%3E");
+                    break;
+                case '?':
+                    sb.append("%3F");
+                    break;
+                case '@':
+                    sb.append("%40");
+                    break;
+                case '[':
+                    sb.append("%5B");
+                    break;
+                case ']':
+                    sb.append("%5D");
+                    break;
+                case '^':
+                    sb.append("%5E");
+                    break;
+                case '`':
+                    sb.append("%60");
+                    break;
+                case '{':
+                    sb.append("%7B");
+                    break;
+                case '|':
+                    sb.append("%7C");
+                    break;
+                case '}':
+                    sb.append("%7D");
+                    break;
+                case '"':   //  %22
+                case '-':   //  %2D
+                case '.':   //  %2E
+                case '/':   //  %2F
+                case ':':   //  %3A
+                case ';':   //  %3B
+                case '\\':  //  %5C
+                case '_':   //  %5F
+                case '~':   //  %7E
+                default:
+                    sb.append(c);
+            }
+
+        }
+
+        return sb.toString();
+
+    }
+
```

Comment:
I see. It would be useful to add that note in the function comment. 
An enthusiastic refactoring in a few years might make the same false assumptions I had and swap it out.